webviz_subsurface/plugins/_co2_leakage/_utilities/co2volume.py

```diff
@@ -283,31 +332,46 @@ def generate_co2_volume_figure(
     df = _read_terminal_co2_volumes(
         table_provider, realizations, scale, containment_info
     )
-    if containment_info["containment_view"] == ContainmentViews.ZONESPLIT:
-        color = "zone"
-        cat_ord = {"zone": containment_info["zones"], "phase": ["gas", "aqueous"]}
-        colors = _split_colors(len(containment_info["zones"]))
-    elif containment_info["containment_view"] == ContainmentViews.REGIONSPLIT:
-        color = "region"
-        cat_ord = {"region": containment_info["regions"], "phase": ["gas", "aqueous"]}
-        colors = _split_colors(len(containment_info["regions"]), "region")
+    view = containment_info["containment_view"]
+    if view != ContainmentViews.CONTAINMENTSPLIT:
+        split = "zone" if view == ContainmentViews.ZONESPLIT else "region"
+        color = "type"
+        pattern_shape = "type"  # ['', '/', '\\', 'x', '-', '|', '+', '.'],
+        cat_ord, colors, pattern = _prepare_pattern_and_color_options(
+            containment_info,
+            split,
+        )
+
+        df["type"] = [
+            ", ".join((zn, cn)) for zn, cn in zip(df[split], df["containment"])
+        ]
     else:
         color = "containment"
         cat_ord = {
             "containment": ["hazardous", "outside", "contained"],
             "phase": ["gas", "aqueous"],
         }
         colors = [_COLOR_HAZARDOUS, _COLOR_OUTSIDE, _COLOR_CONTAINED]
+        pattern_shape = "phase"
+        pattern = ["", "/"]
+    df["prop"] = np.zeros(df.shape[0])
+    for r in realizations:
+        summed_amount = np.sum(df.loc[df["real"] == str(r)]["amount"])
+        df.loc[df["real"] == str(r), "prop"] = summed_amount
+    df["prop"] = np.round(df["amount"] / df["prop"] * 1000) / 10
+    df["prop"] = df["prop"].map(lambda p: str(p) + "%")
```

Comment:
Burde vi håndtere å ikke dele på null her? Eller er alltid det vi deler på totalen i siste dato for en gitt realisasjon, så det skal veldig mye til at de er 0?

Reply:
Godt poeng, det har jeg glemt

Reply:
Kan prøve hvis det ikke tar for lang tid